core/controllers/ingest.py

```diff
@@ -361,3 +364,34 @@ def remove_unreferenced_organisations():
             db.session.delete(organisation)
 
     db.session.commit()
+
+
+def with_transaction(
+    function: callable, max_retries: int, sleep_duration: float, error_type: Type[Exception], **kwargs: dict
+) -> None:
+    """Execute a transaction with retries for a specified error type.
+
+    This function wraps the provided 'function' in a database transaction and allows for retries
```

Comment:
Can we just re-word the first line of summary paragraph slightly, as "wrap" is a bit misleading in Python. So something like, "Run provided function within an atomic database transaction..."

Reply:
Which makes me think, could we implement this function as a decorator? Might be useful? Or is that overkill?

Reply:
It means every time we want to run a function within a transaction/retry loop, we just decorate it in place, rather than always nesting within another function. Any thoughts?